cmake/CMakeLists.txt

```diff
@@ -1273,7 +1273,7 @@ else()
 endif()
 
 if (WIN32)
-  if (onnxruntime_BUILD_FOR_WINDOWS_STORE)
+  if (WINDOWS_STORE)
```

Comment:
What is the main purpose of checking for WINDOWS_STORE instead of onnxruntime_BUILD_FOR_WINDOWS_STORE?

Reply:
Previously WINDOWS_STORE would be enabled in store and non-store to always link to windowsapp.lib, and onnxruntime_BUILD_FOR_WINDOWS_STORE was our own thing to differentiate whether we really want store or just the runtime workaround. I'm removing the workaround altogether and keeping only WINDOWS_STORE